projects/ids-enterprise-ng/src/lib/alert/soho-alert-directive.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, DebugElement, ViewChild } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { SohoAlertDirective } from './soho-alert.directive';
+import { SohoAlertModule } from './soho-alert.module';
+
+@Component({
+  template: `
+    <div class="field">
+      <input soho-alert [message]="errorMsg">
+    </div>
+    `
+})
+class SohoAlertDirectiveTestComponent {
+  @ViewChild(SohoAlertDirective) alert?: SohoAlertDirective | null;
+  errorMsg = '';
+}
+
+describe('Soho Alert Directive Render', () => {
+  let alert: SohoAlertDirective;
+  let comp: SohoAlertDirectiveTestComponent;
+  let fixture: ComponentFixture<SohoAlertDirectiveTestComponent>;
+  let de: DebugElement;
+  let el: HTMLElement;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [SohoAlertDirectiveTestComponent],
+      imports: [SohoAlertModule],
+    });
+
+    fixture = TestBed.createComponent(SohoAlertDirectiveTestComponent);
+    comp = fixture.componentInstance;
+
+    de = fixture.debugElement;
+    el = de.query(By.css('input[soho-alert]')).nativeElement;
+
+    fixture.detectChanges();
+    alert = (comp.alert as any);
+    (alert as any)._options.type = 'error';
```

Comment:
does it mean as an end user, when I use `soho-alert`, I need to do
```
<input soho-alert [type]="'error'" />
```
so that setting `[message]="''"` would clear the error message?

Reply:
Hello, I tried it and there shouldn't be a need to set a type if you set the message that way, the default will be error. I think it's only when you don't use the public methods that problems might happen, _options is private

Reply:
I added the repo to reproduce the issue in the bug issue.

Reply:
So this line should not be added in this unit test because the unit test is correct. The directive has a bug.

Reply:
Hello, thank you for the sample app, it helped in making me understand the bug better. I pushed some changes in this branch, your original tests should work now